src/PlaywrightSharp.Firefox/FirefoxBrowserContext.cs

```diff
@@ -56,13 +64,33 @@ public async Task CloseAsync()
             _browser.Contexts.Remove(_browserContextId);
         }
 
+        /// <inheritdoc cref="IBrowserContextDelegate.GetExistingPages"/>
         public IEnumerable<IPage> GetExistingPages() => _browser.GetAllTargets()
                 .Where(target => target.BrowserContext == BrowserContext && target.Page != null)
                 .Select(target => target.Page);
 
+        /// <inheritdoc cref="IBrowserContextDelegate.SetPermissionsAsync(string, ContextPermission[])"/>
         public Task SetPermissionsAsync(string origin, params ContextPermission[] permissions)
         {
             throw new System.NotImplementedException();
         }
+
+        /// <inheritdoc cref="IBrowserContextDelegate.SetCookiesAsync(SetNetworkCookieParam[])"/>
+        public Task SetCookiesAsync(params SetNetworkCookieParam[] cookies)
+            => _connection.SendAsync(new BrowserSetCookiesRequest
+            {
+                BrowserContextId = _browserContextId,
+                Cookies = Array.ConvertAll(cookies, c => (CookieOptions)c),
+            });
+
+        /// <inheritdoc cref="IBrowserContextDelegate.GetCookiesAsync"/>
+        public async Task<NetworkCookie[]> GetCookiesAsync()
+        {
+            var result = await _connection.SendAsync(new BrowserGetCookiesRequest { BrowserContextId = _browserContextId }).ConfigureAwait(false);
+            return Array.ConvertAll(result.Cookies, c => (NetworkCookie)c);

```

Comment:
```suggestion
            return result.Cookies.Cast<NetworkCookie>();
```

And we could return an `IEnumerable` so we don't cast all of them if we don't need to.